src/main/java/de/tum/cit/aet/artemis/communication/service/ReactionService.java

```diff
@@ -195,4 +199,34 @@ private Reaction createReactionForPost(Reaction reaction, Post posting, User use
         api.broadcastForPost(new PostDTO(post, MetisCrudAction.UPDATE), course.getId(), null);
         return savedReaction;
     }
+
+    /**
+     * Returns the course for the given reaction, throws if not found.
+     *
+     * @param reaction the reaction entity
+     * @return the associated course
+     * @throws BadRequestAlertException if no course can be found
+     */
+    private Course getReactionCourseElseThrow(Reaction reaction) {
+        if (reaction.getPost() != null) {
+            return reaction.getPost().getCoursePostingBelongsTo();
+        }
+        if (reaction.getAnswerPost() != null) {
+            return reaction.getAnswerPost().getCoursePostingBelongsTo();
+        }
+        throw new BadRequestAlertException("Reaction could not be found", METIS_REACTION_ENTITY_NAME, "reactionNotFound");
+    }
+
+    /**
+     * Ensures that the given course matches the expected course ID.
+     *
+     * @param expectedCourseId the ID of the course that is expected
+     * @param actual           the actual course to verify
+     * @throws BadRequestAlertException if the actual course is null or does not match the expected course ID
+     */
+    private void assertBelongsToCourse(Long expectedCourseId, Course actual) {
```

Comment:
```suggestion
    private void checkThatCourseHasCourseIdElseThrow(Long expectedCourseId, Course actual) {
```
I would prefer this name. I think about tests when I hear assert.